apps/ecommerce/lambda/src/controllers/CredentialsController.ts

```diff
@@ -0,0 +1,42 @@
+import axios, { AxiosError, AxiosResponse } from 'axios';
+import { Request, Response, NextFunction, Router } from 'express';
+
+const CredentialsController = Router();
+
+// TODO: Refactor this so that it pulls in the headers values (shopify storefront token, storename, etc...) separately
+const checkCredentials = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    if (!req.appConfig) throw new Error('App config not found');
+
+    const proxyResourceUrl = new URL(req.appConfig.baseUrl);
+    proxyResourceUrl.pathname += `/credentials`;
+
+    try {
+      let response;
+      try {
+        response = await axios.get(proxyResourceUrl.toString(), {
+          headers: {
+            'x-data-provider-parameters': JSON.stringify(req.installationParameters),
```

Comment:
This is super easy to follow so far, great job! I may be misunderstanding some of this logic or perhaps it is a follow-up item, but are we passing the installation parameters as part of the request, within `x-data-provider-parameters`? If so, why are we not just using the CMA within the lambda to pull those parameters without any communication with the frontend?

Reply:
Excellent thought/callout.

This is because when you click Save on the ConfigPage those parameters aren't set yet. So lets say you want to do some validation checking and then provide an error message like we did in GA4 using the `onConfigure` handler, then you need to pass these installation parameters as headers because they don't even exist as installation parameters yet!

Reply:
ah okay so this is prior to the user saving their config! gotcha! trying to visualize how this manifesting right now, in what workflow are we eagerly rendering anything specific if the user has not yet saved the config?